price-estimator/src/orderbook.rs

```diff
@@ -52,34 +53,41 @@ impl Orderbook {
     pub async fn pricegraph(
         &self,
         time: EstimationTime,
+        ignore_addresses: &[Address],
         pricegraph_type: PricegraphKind,
     ) -> Result<Pricegraph> {
-        match time {
-            EstimationTime::Now => Ok(self.cached_pricegraph(pricegraph_type).await),
-            EstimationTime::Batch(batch_id) => {
-                let mut auction_data = self.auction_data(batch_id).await?;
-                if matches!(pricegraph_type, PricegraphKind::WithRoundingBuffer) {
-                    self.apply_rounding_buffer_to_auction_data(&mut auction_data)
-                        .await?;
-                }
-                Ok(pricegraph_from_auction_data(&auction_data))
+        let mut auction_data = match (time, ignore_addresses.is_empty()) {
+            (EstimationTime::Now, true) => {
+                return Ok(self.cached_pricegraph(pricegraph_type).await)
```

Comment:
I think it will be important for the unimplemented branch too so prefer like this.

Reply:
I slightly prefer it as it is as well, since it reduces rightward drift. I also expect the compiler to optimize the `is_empty()` call into the `EstimationTime::Now` branch so I don't think it should actually make any difference.